action/governance/fundProposal.go

```diff
@@ -0,0 +1,211 @@
+}
+
+func runFundProposal(ctx *action.Context, tx action.RawTx) (bool, action.Response) {
+	fundProposal := FundProposal{}
+	err := fundProposal.Unmarshal(tx.Data)
+	if err != nil {
+		return false, action.Response{}
+	}
+	//1. check if proposal exists
+	proposal, err := ctx.ProposalMasterStore.Proposal.Get(fundProposal.ProposalId)
+	if err != nil {
+		ctx.Logger.Error("Proposal does not exist :", fundProposal.ProposalId)
+		result := action.Response{
+			Events: action.GetEvent(fundProposal.Tags(), "fund_proposal_does_not_exist"),
+		}
+		return false, result
+	}
+
+	//2. Check if the Funding height is already reached
+	//  If the proposal has already passed Funding height, reject the transaction
+	if ctx.Header.Height > proposal.FundingDeadline {
+		ctx.Logger.Debug("Funding Height has already been reached")
+		result := action.Response{
+			Events: action.GetEvent(fundProposal.Tags(), "fund_proposal_height_crossed"),
+		}
+		return false, result
+
+	}
+	//3. Check if the Proposal is in funding stage
+	//  If the proposal is not in FUNDING state, reject the transaction
+	if proposal.Status != governance.ProposalStatusFunding {
+		ctx.Logger.Debug("Cannot fund proposal , Current proposal state : ", proposal.Status)
+		result := action.Response{
+			Events: action.GetEvent(fundProposal.Tags(), "fund_proposal_not_funding_state"),
+		}
+		return false, result
+	}
+
+	//4. Check if the Proposal has reached funding goal, when this contribution is added
+	//   Change the state of the proposal to VOTING, if funding goal is met
+	fundingAmount := balance.NewAmountFromBigInt(fundProposal.FundValue.Value.BigInt())
+	currentFundsforProposal := governance.GetCurrentFunds(proposal.ProposalID, ctx.ProposalMasterStore.ProposalFund)
+	newAmount := fundingAmount.Plus(currentFundsforProposal)
+	if newAmount.BigInt().Cmp(proposal.FundingGoal.BigInt()) >= 0 {
+		//5. Update status and set voting deadline
+		proposal.Status = governance.ProposalStatusVoting
+		options := ctx.ProposalMasterStore.Proposal.GetOptionsByType(proposal.Type)
+		proposal.VotingDeadline = ctx.Header.Height + options.VotingDeadline
+
+		//6. If the proposal moves into Voting state, take a snap shot of Validator Set,
+		//at that instant and add entries for each and every validator at the point into the Proposal_Vote_Store.
+		//In the value, we will just update the Voting power. The vote / opinion field remains empty for now
+		validatorList, err := ctx.Validators.GetValidatorSet()
+		if err != nil {
+			return false, action.Response{Log: "Fund proposal failed in getting validator list"}
+		}
+		for _, v := range validatorList {
+			vote := governance.NewProposalVote(v.Address, governance.OPIN_UNKNOWN, v.Power)
+			err = ctx.ProposalMasterStore.ProposalVote.Setup(proposal.ProposalID, vote)
+			if err != nil {
+				return false, action.Response{Log: "setup voting validator failed"}
+			}
+		}
+
+		//7. Update proposal status to VOTING
+		err = ctx.ProposalMasterStore.Proposal.Set(proposal)
```

Comment:
Can you make sure you are setting the proposal to the active prefix?

Reply:
This change is already there in my branch ,maybe I can put a comment to check it when we merge my branch to develop